ingestion-beam/src/main/java/com/mozilla/telemetry/util/Normalize.java

```diff
@@ -39,4 +65,64 @@ public static String channel(String name) {
       return OTHER;
     }
   }
+
+  /**
+   * Normalize an operating system string to one of the three major desktop platforms,
+   * one of the two major mobile platforms, or "Other".
+   *
+   * <p>Reimplements and combines the logic of {@code os} and {@code mobile_os} from
```

Comment:
We combine the desktop and mobile here to support the [propose common metadata structure](https://docs.google.com/document/d/1AsshL_QS1qRVXcHW5lz0jZeM-UKIWFl1oV3XJz7Xewo/edit?disco=AAAACsG0pZA&usp_dm=false&ts=5cb4b194). All tables loaded by the ingestion pipeline will have a `normalized_os` field.